src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -0,0 +1,274 @@
+(* this is ugly, but [Core_kernel.String] doesn't have a [trim] function *)
+let trim = String.trim
+
+open Core_kernel
+
+(* instantiate snarky for Vesta *)
+
+module Impl = Snarky_backendless.Snark.Run.Make (struct
+  include Kimchi_backend.Pasta.Vesta_based_plonk
+  module Inner_curve = Kimchi_backend.Pasta.Pasta.Pallas
+end)
+
+(* helpers *)
+
+let compare_with obtained filepath =
+  let (_ : int) = Sys.command "tree" in
+  let filepath = "examples/" ^ filepath in
+  let expected = In_channel.read_all filepath in
+  if String.(trim obtained <> trim expected) then (
+    Format.printf "mismatch for %s detected\n" filepath ;
+    Format.printf "expected:\n%s\n\n" expected ;
+    Format.printf "obtained:\n%s\n" obtained ;
+    failwith "circuit compilation has changed" )
+
+let check_json ~input_typ ~return_typ ~circuit filename () =
+  let cs : Impl.R1CS_constraint_system.t =
+    Impl.constraint_system ~input_typ ~return_typ circuit
+  in
+  let serialized_json =
+    Kimchi_backend.Pasta.Vesta_based_plonk.R1CS_constraint_system.to_json cs
+  in
+  compare_with serialized_json filename
+
+(* monadic API tests *)
+
+(** Both the monadic and imperative API will produce the same circuit hash. *)
+let expected = "5357346d161dcccaa547c7999b8148db"
+
+module MonadicAPI = struct
+  module Impl = Snarky_backendless.Snark.Make (struct
+    include Kimchi_backend.Pasta.Vesta_based_plonk
+    module Inner_curve = Kimchi_backend.Pasta.Pasta.Pallas
+  end)
+
+  let main ((b1, b2) : Impl.Boolean.var * Impl.Boolean.var) =
```

Comment:
It's not clear what this is supposed to be testing.. some boolean stuff?

Reply:
this is just a simple circuit to make sure that it hashes to the same digest when compiled using two different APIs